google/resource_dataproc_cluster.go

```diff
@@ -0,0 +1,870 @@
+
+						"preemptible_boot_disk_size_gb": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+							ForceNew: true,
+							ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+								value := v.(int)
+
+								if value < 10 {
+									errors = append(errors, fmt.Errorf(
+										"%q cannot be less than 10", k))
+								}
+								return
+							},
+						},
+					},
+				},
+			},
+
+			"service_account": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"service_account_scopes": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					StateFunc: func(v interface{}) string {
+						return canonicalizeServiceScope(v.(string))
+					},
+				},
+			},
+
+			"initialization_action_timeout_sec": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"initialization_actions": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: false,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"metadata": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: true,
+				Elem:     schema.TypeString,
+			},
+
+			"master_instance_names": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"worker_instance_names": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"preemptible_instance_names": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceDataprocClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	timeoutInMinutes := int(d.Timeout(schema.TimeoutCreate).Minutes())
+
+	// Mandatory
+	clusterName := d.Get("name").(string)
+	region := d.Get("region").(string)
+	zone, zok := d.GetOk("zone")
+
+	if region == "global" && !zok {
+		return errors.New("zone is mandatory when region is set to 'global'")
+	}
+
+	gceConfig := &dataproc.GceClusterConfig{}
+
+	if v, ok := d.GetOk("network"); ok {
+		gceConfig.NetworkUri = extractLastResourceFromUri(v.(string))
+	}
+
+	if v, ok := d.GetOk("subnetwork"); ok {
+		gceConfig.SubnetworkUri = extractLastResourceFromUri(v.(string))
+	}
+
+	if v, ok := d.GetOk("tags"); ok {
+		tagsList := v.([]interface{})
+		tags := []string{}
+		for _, v := range tagsList {
+			tags = append(tags, v.(string))
+		}
+		gceConfig.Tags = tags
+	}
+
+	if v, ok := d.GetOk("service_account"); ok {
+		gceConfig.ServiceAccount = v.(string)
+	}
+
+	if v, ok := d.GetOk("service_account_scopes"); ok {
+		scopesList := v.([]interface{})
```

Comment:
This one is slightly different as it also does a transformation on the string. That said, created another helper function `convertAndMapStringArr` to do similar